### PR TITLE
adds polymorphism to the application status api

### PR DESCRIPTION
**Pull request by daynekilheffer**
Uses `discriminator` ([docs](https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/)) to introduce polymorphism to the application status api.

### CODE REVIEW

**Review by kevinrathgeber — APPROVED** [review]

File: scoir-api.yaml [hunk 1]
```diff
@@ -112,16 +112,14 @@ paths:
...
-                      $ref: "#/components/schemas/scoir_application_status"
-                  otherApplications:
-                    description: Application not originated on Scoir.  
-                    type: array
-                    items:
-                      $ref: "#/components/schemas/other_application_status"
+                      anyOf:
+                        - $ref: "#/components/schemas/scoir_application_status"
+                        - $ref: "#/components/schemas/other_application_status"
+                      discriminator:
```
Comment (dags2112) [inline scoir-api.yaml:121]: I might be off here but I feel like using the scoirApplicationId as the discriminator isn't quite right, shouldn't this be able to map to one of the types in the anyOf: array?

From the next section of their docs:
`It is implied, that the property to which discriminator refers, contains the name of the target schema. In the example above, the objectType property should contain either simpleObject, or complexObject string. If the property values do not match the schema names, you can map the values to the names. To do this, use the discriminator/mapping keyword:`

Comment (daynekilheffer) [reply scoir-api.yaml:121]: You are correct; oversight on my part.

Comment (daynekilheffer) [reply scoir-api.yaml:121]: you are correct - misunderstanding on my part. What are your thoughts on this change if that is the case?

Comment (dags2112) [reply scoir-api.yaml:121]: depends on what our goal is with this, are we trying to create one endpoint that can take a mixed array of scoir apps, and other app types? looks like the endpoint def specifies two array params w/ different types in them

i might be missing the point here but my naive thought is just to add an applicationType field to map to different schemas or even just have a separate endpoint

let me know if you want to chat through this, not sure, i am conveying my thoughts well here

File: schemas/applicationStatus.yml [hunk 2]
```diff
@@ -12,9 +12,15 @@ components:
         scoirApplicationId:
           type: string
           description: "Unique identifier assigned to a college application initiated on Scoir."
-        externalSSOId:
+        applicationType:
```
Comment (dags2112) [inline schemas/applicationStatus.yml:15]: as i pushed this i started to question the name, and type, maybe this should just be a bool named scoirInitiated, applicationType feels more like Coalition vs Common App vs Institutional App etc @daynekilheffer @kevinrathgeber @aowen thoughts?

Comment (daynekilheffer) [reply schemas/applicationStatus.yml:15]: if it's not coming together naturally, maybe discriminator is just a bad idea.  We can use `oneOf`/`anyOf` without it

File: schemas/applicationStatus.yml [hunk 3]
```diff
@@ -12,9 +12,15 @@ components:
...
           type: string
-          description: "Unique identifier assigned by the Application Platform that allows user to SSO into the Application Platform."
+          enum:
+            - scoir
+            - other
+          description: "Indicates if the application was intitated through Scoir vs another Application Platform."
+        externalId:
```
Comment (dags2112) [inline schemas/applicationStatus.yml:21]: @aowen @daynekilheffer @scoir-alexr I renamed this to just externalId, externalSSOId felt like an implementation detail that might not always apply, after application submission the third party should already have all the scoir ids/details they'd need to process an SSO request based on the identity info we pass along with it, might be missing something here but we may not need this id at all? maybe we should sync up on this?

Comment (aowen) [reply schemas/applicationStatus.yml:21]: yeah, let's please sync on this before we make the decision to rename `externalSSOId`

Comment (aowen) [reply schemas/applicationStatus.yml:21]: nevermind...i just rendered this PR in ReDoc to wrap my head around these changes.  i like them all!

**Review by aowen — APPROVED** [review]

**Pull request merged**